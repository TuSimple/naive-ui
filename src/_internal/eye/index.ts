export { default } from './src/Eye'
