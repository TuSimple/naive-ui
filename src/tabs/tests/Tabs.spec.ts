import { mount } from '@vue/test-utils'
import { h } from 'vue'
import { NTabPane, NTabs } from '../index'
import { AddIcon } from '../../_internal/icons'

describe('n-tabs', () => {
  it('should work with import on demand', () => {
    mount(NTabs)
  })
  it('should work with callback types', () => {
    function onUpdateValue1 (name: number): void {}
    function onUpdateValue2 (name: string): void {}
    function onUpdateValue3 (name: number | string): void {}
    mount(NTabs, {
      props: {
        onUpdateValue: onUpdateValue1
      }
    })
    mount(NTabs, {
      props: {
        onUpdateValue: onUpdateValue2
      }
    })
    mount(NTabs, {
      props: {
        onUpdateValue: onUpdateValue3
      }
    })
  })
  it('should work with empty tab-pane', () => {
    mount(NTabs, {
      props: {
        defaultValue: 'a'
      },
      slots: {
        default: () =>
          h(NTabPane, {
            tab: 'a',
            name: 'a'
          })
      }
    })
  })

  it('should show AddIcon with `addable` prop', () => {
    const wrapper = mount(NTabs, {
      props: {
        type: 'card',
        addable: true
      }
    })

    expect(wrapper.findComponent(AddIcon).exists()).toBe(true)
  })

  it('should work with `justify-content` prop', async () => {
    const wrapper = mount(NTabs, {
      props: {
        defaultValue: 'a'
      }
    })

    await wrapper.setProps({ justifyContent: 'space-between' })
    expect(
      wrapper.find('.n-tabs-wrapper').attributes('style')
    ).toMatchSnapshot()

    await wrapper.setProps({ justifyContent: 'space-around' })
    expect(
      wrapper.find('.n-tabs-wrapper').attributes('style')
    ).toMatchSnapshot()

    await wrapper.setProps({ justifyContent: 'space-evenly' })
    expect(
      wrapper.find('.n-tabs-wrapper').attributes('style')
    ).toMatchSnapshot()
  })
})
