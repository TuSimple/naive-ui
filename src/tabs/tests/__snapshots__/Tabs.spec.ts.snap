// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`n-tabs should work with \`justify-content\` prop 1`] = `"display: flex; justify-content: space-between;"`;

exports[`n-tabs should work with \`justify-content\` prop 2`] = `"display: flex; justify-content: space-around;"`;

exports[`n-tabs should work with \`justify-content\` prop 3`] = `"display: flex; justify-content: space-evenly;"`;
